### PR TITLE
Expose some more functions from libc

### PR DESCRIPTION
**Pull request by achanda**
https://github.com/rust-lang/rust/issues/18271

I noticed this issue was moved to the rfc repo after I opened the PR. Should this be closed now and prioritized later?

### CODE REVIEW

**Comment by brson** [comment]
r? @aturon 

I don't know whether these are desired - some of them have definitely been removed in the past. At the least they need be marked unstable before entering the tree - you can see the unstable attribute syntax elsewhere in std.

File: src/libstd/num/f32.rs [hunk 1]
```diff
@@ -680,6 +732,21 @@ impl f32 {
...
+    /// ```
+    /// let x = 5.0f32;
+    /// let y = 2.0f32;
+    ///
+    /// assert_eq!(x.rem(y), 1.0);
+    /// ```
+    #[unstable(feature = "std_misc",
+                reason = "unsure about its place in the world")]
+    #[inline]
+    pub fn rem(self, other: f32) -> f32 {
```
Comment (alexcrichton) [inline src/libstd/num/f32.rs]: I believe that this is already exposed through the `Rem` trait (LLVM lowers `%` operations to calls to this function I believe).

File: src/libstd/num/f32.rs [hunk 2]
```diff
@@ -652,6 +652,58 @@ impl f32 {
...
+        unsafe { cmath::logbf(self) }
+    }
+
+    /// Returns integer exponent of the input
+    ///
+    /// ```
+    /// use std::f32;
+    ///
+    /// let x = 0.5f32;
+    /// let abs_diff = (x.logb() - 1.0f32).abs();
```
Comment (alexcrichton) [inline src/libstd/num/f32.rs]: This is perhaps an example for a different function?

**Comment by alexcrichton** [comment]
This is an area where we may not want to follow the conventional C names and instead choose a more rustic/descriptive name as well.

**Comment by achanda** [comment]
Thanks @alexcrichton and @brson . I'm totally open to suggestions on names.

File: src/libstd/num/f32.rs [hunk 3]
```diff
@@ -652,6 +652,58 @@ impl f32 {
...
+    ///
+    /// ```
+    /// use std::f32;
+    ///
+    /// let x = 0.5f32;
+    /// let abs_diff = (x.logb() - 1).abs();
+    ///
+    /// assert!(abs_diff <= f32::EPSILON)
+    /// ```
+    #[unstable(feature = "std_misc",
```
Comment (alexcrichton) [inline src/libstd/num/f32.rs]: Could these functions also be introduced under a new feature name? We're trying to cut down on the number of members of the `std_misc` feature, and something like `float_extras` may work well here.

File: src/libstd/num/f64.rs [hunk 4]
```diff
@@ -628,6 +628,58 @@ impl f64 {
...
+    /// Returns the value of the gamma function
+    /// at the given point
+    /// ```
+    /// use std::f64;
+    ///
+    /// let x = 1.2f64;
+    /// let g = 0.9182f64;
+    /// let abs_diff = (x.gamma() -f ).abs();
+    ///
+    /// assert!(abs_diff <= f64::EPSIOLON)
```
Comment (alexcrichton) [inline src/libstd/num/f64.rs]: I think that EPSILON here may be misspelled, and you can also be sure to run these tests via:

```
make check-stage1-doc-crate-std
```

**Comment by aturon** [comment]
FWIW, the names of the methods currently available are a bit of a mixed bag. For the most straightforward cases, we tried to hew toward Rust-style names, but for more obscure things like `exp_m1` we stuck with the libc names, for familiarity sake. I'm... not really sure what's best for these proposed additions, whose names inherited names are pretty bad! Can we think of something more descriptive?

**Comment by achanda** [comment]
Thanks all. The doc tests are not passing yet, working on it.

**Comment by alexcrichton** [comment]
@bors: r+ 02a188b1dd31c16278a35667ad5e9303f795c215

Thanks @achanda!

**Comment by achanda** [comment]
@alexcrichton the doc tests were not passing when I last tested this on my box. I did not have time to debug this further. Is it okay to merge in this state?

**Comment by alexcrichton** [comment]
@bors: r-

Ah no those will need to be fixed before merging.

**Comment by achanda** [comment]
Guessed so. Please give me some time to get back to this.

**Comment by alexcrichton** [comment]
We actually had some time to discuss this in triage today and the conclusion was actually to not merge this at this time. The current bindings to libm are unfortunately not the best thought out and it's not clear that we actually want to continue providing them. At this time we think that stemming the flow of new bindings to libm is the best route to take for now while we take some time to work out a concrete story for the functions here. I'll open an associated issue for this, but for now I'm going to close this. Sorry for the delay!

**Pull request closed without merging**